docs/bls-keys/dd.md

```diff
@@ -0,0 +1,97 @@
+
+## Data types
+
+We add a `BLSPublicKey` type, backed 48 bytes data, to fit a public key on the BLS12-381 curve. Along with this type, we add a method for validating that a given `BLSPublicKey` instance is a valid point in the curve.
+
+## Interface changes
+
+The following methods or structs require an additional `BLSPublicKey` field:
+
+- `StakingQueue.enqueue`
+- `StakingQueue.DepositArgs`
+- `IStakingCore.deposit`
+- `StakingLib.deposit`
+- `RollupCore.deposit`
+- `ExtRollupLib2.deposit`
+- `GSE.deposit`
+- `GSE.AttesterConfig`
+
+All GSE functions that return a set of attester addresses now require an additional function for fetching their BLS keys:
+
+- `GSE.getAttestersFromIndicesAtTime` -> `GSE.getAttestersBLSPubKeysFromIndicesAtTime`
+- `GSE.getAttestersAtTime` -> `GSE.getAttestersBLSPubKeysAtTime`
+- `GSE.getAttesterFromIndexAtTime` -> `GSE.getAttesterBLSPubKeyFromIndexAtTime`
+
+## Security considerations
+
+When registering a new key, we first need to validate that it is a correct public key in the curve. Then, as described in [this article](https://www.zellic.io/blog/bls-signature-versatility/#the-pitfall-of-multi-signatures), we need to guard against rogue key attacks. We have two options for that:
+
+1. Require a proof of possession, validated along with the key during registration
+2. Use a modified aggregate public key
+
+Given the second option has not been implemented (at least based on the article linked above), and that it may involve higher gas costs for generating the aggregate public key (which happens on every epoch) as opposed to the one-time registration cost, the first option seems the best.
+
+## Open questions
+
+### Is the choice of curve correct?
+
+We defaulted to the curve used by ZCash and Ethereum consensus, but there may be other curves that are a better fit.
+
+### When should we validate the public key is valid?
+
+The innermost contract (GSE) seems to be the most secure place where to execute this validation.
+
+On the other hand, if we know that all entries to GSE have to go through the staking queue, we could instead validate early in the `enqueue`, which would also allow us to change curves in the future without having to modify the GSE.
+
+To favor failing fast, I prefer validating in the queue, assuming we are certain that the queue is always present.
+
+### How to prevent against the rogue key attack?
+
+As discussed above in the Security Considerations section, we need to decide between requiring a proof of possession or using a modified aggregate public key.
+
+Given existing implementations and gas costs, I vote for the proof of possession.
+
+### How do we generate the BLS private key for validators?
+
+We could derive the BLS private key from their ECDSA private key, potentially from having them sign a given message so we do not need direct access to the ECDSA private key. This means that validators do not need to store two separate private keys.
```

Comment:
deterministic ECDSA should be a consideration here. It has caused problems in the past, where lattice plus did not implement deterministic nonces

Reply:
if using the same private key it would need to be reduced to the field size of BLS12-381